protocol/ClientPackets.cpp

```diff
@@ -21,4 +21,24 @@ std::shared_ptr<std::vector<uint8_t>> protocol::createStatusResponse(const Statu
     auto packet = std::make_shared<std::vector<uint8_t>>();
     finalize(*packet.get(), payload, (int32_t)ClientPacketID::STATUS);
     return packet;
-}
\ No newline at end of file
+}
+
+std::shared_ptr<std::vector<uint8_t>> protocol::createPlayerChatMessage(const PlayerChatMessage &in)
+{
+    std::vector<uint8_t> payload;
+    serialize(payload, in.signedContent, addString);
+    serialize(payload, in.hasUnsignedContent, addBoolean);
+    serialize(payload, in.unsignedContent, addString);
+    serialize(payload, in.type, addVarInt);
+    serialize(payload, in.senderUUID, addUUID);
+    serialize(payload, in.senderName, addString);
+    serialize(payload, in.hasTeamName, addBoolean);
+    serialize(payload, in.teamName, addString);
+    serialize(payload, in.timestamp, addLong);
+    serialize(payload, in.salt, addLong);
+    serialize(payload, in.signatureLength, addVarInt);
+    serialize(payload, in.signature, addByteArray);
```

Comment:
Please only use one call to `serialize`, it has a variable number of arguments like `parse`